trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/ACSLHandler.java

```diff
@@ -751,13 +751,15 @@ public Result visit(final IDispatcher main, final QuantifierExpression node) {
 	@Override
 	public Result visit(final IDispatcher main, final Contract node) {
 		final ILocation loc = mLocationFactory.createACSLLocation(node);
-		final ArrayList<Specification> spec = new ArrayList<>();
+
+		// TODO The check below is broken. The compiler detects this and reports an error.
 		// First we catch the case that a contract is at a FunctionDefinition
-		if (node instanceof IASTFunctionDefinition) {
-			final String msg = "Syntax Error, Contracts on FunctionDefinition are not allowed";
-			throw new IncorrectSyntaxException(loc, msg);
-		}
+		// if (node instanceof IASTFunctionDefinition) {
+		// final String msg = "Syntax Error, Contracts on FunctionDefinition are not allowed";
+		// throw new IncorrectSyntaxException(loc, msg);
+		// }

```

Comment:
Making the type hierarchy sealed causes the compiler to detect that this  `instanceof` check  is meaningless (i.e., always `false`) and report an error. The check was always meaningless, it's only now being reported.

Can this check be simply removed? If not, where could this be checked instead? The problem with checking it here is that we need the C AST node that is annotated with a `Contract`, but we do not have it here (and we accidentally used the ACSL AST node instead).

Reply:
I think this check should be removed. It seems like the intention was to crash on ACSL contracts at function definitions, but this should be supported by now (e.g. see [ACSL-contractOnDelarationAndDefinition.c](https://github.com/ultimate-pa/ultimate/blob/dev/trunk/examples/programs/regression/c/ACSL-contractOnDelarationAndDefinition.c)).